kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -171,12 +174,38 @@ public void forwardResponse(ApiMessage response) {
         if (!name.endsWith("ResponseData")) {
             throw new AssertionError("Attempt to use forwardResponse with a non-response: " + name);
         }
-        // TODO check we've not forwarded it already
+        if (decodedFrame instanceof RequestFrame) {
```

Comment:
This PR should improve the documentation in the KrpcFilterContext#forwardResponse to make the new use-case clear in the javadoc.